src/borg/testsuite/repository.py

```diff
@@ -15,11 +13,21 @@
 from ..remote import RemoteRepository, InvalidRPCMethod, PathNotAllowed
 from ..repository import Repository, LoggedIO, MAGIC, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
 from ..repoobj import RepoObj
-from . import BaseTestCase
 from .hashindex import H
 
 
-UNSPECIFIED = object()  # for default values where we can't use None
+@pytest.fixture()
+def repository(tmpdir):
+    repository_location = os.path.join(str(tmpdir), "repository")
+    repository = Repository(repository_location, exclusive=True, create=True)
+    yield repository
```

Comment:
If it wasn't for needing to reopen the repository during a few tests, I would have done something like:
```
with Repository(repository_location, exclusive=True, create=True) as repository
     yield repository
```
since this CM would automatically enter and exit the repository context. But I can't figure out how to get `reopen()` to work with this.